autox/informer.go

```diff
@@ -3,36 +3,146 @@ package autox
 // informer.go - informer(s) to watch desired resources/namespaces
 
 import (
+	"fmt"
+	"reflect"
+
 	"github.com/iter8-tools/iter8/base/log"
-	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/client-go/dynamic/dynamicinformer"
 	"k8s.io/client-go/tools/cache"
 )
 
+const (
+	autoxLabel = "iter8.tools/autox-group"
+)
+
+type chartAction int64
+
+const (
+	releaseAction chartAction = 0
+	deleteAction  chartAction = 1
+)
+
+func getReleaseName(chartGroupName string, chartName string) string {
+	return fmt.Sprintf("autox-%s-%s", chartGroupName, chartName)
+}
+
+func installHelmReleases(resourceName string) {
+	doChartAction(resourceName, releaseAction)
+}
+
+func installHelmRelease(releaseName string, chart chart) {
+	// TODO: check if there is a preexisting Helm release
+
+	// TODO: mutex
+
+	// TODO: release Helm chart
+
+	log.Logger.Debug("Release chart:", releaseName)
+}
+
+func deleteHelmReleases(resourceName string) {
+	doChartAction(resourceName, deleteAction)
+}
+
+func deleteHelmRelease(releaseName string, chart chart) {
+	// TODO: check if there is a preexisting Helm release
+
+	// TODO: mutex
+
+	// TODO: delete Helm chart
+
+	log.Logger.Debug("Delete chart:", releaseName)
+}
+
+func doChartAction(resourceName string, chartAction chartAction) {
+	if cg, ok := iter8ChartGroupConfig[resourceName]; ok {
+		for chartName, chart := range cg.Charts {
+			releaseName := getReleaseName(resourceName, chartName)
+
+			switch chartAction {
+			case releaseAction:
+				installHelmRelease(releaseName, chart)
+
+			case deleteAction:
+				deleteHelmRelease(releaseName, chart)
+			}
+		}
+	} else {
+		// TODO: what log level should this be?
+		log.Logger.Debug("AutoX should make a Helm release for app \"", resourceName, "\" but no Helm charts were provided in the chartGroupConfig")
+	}
+}
+
 var addObject = func(obj interface{}) {
 	log.Logger.Debug("Add:", obj)
+
+	uObj := obj.(*unstructured.Unstructured)
+	resourceName := uObj.GetName()
+	// example label: iter8.tools/autox-group=hello
+	labels := uObj.GetLabels()
+
+	// check if the app name matches the name in the autox label
+	if autoxLabelName, ok := labels[autoxLabel]; ok && resourceName == autoxLabelName {
```

Comment:
I'm confused by why add and delete use this approah and update uses prunelables and compare. I would have thought it would be the same

Reply:
You are right. I should have also updated the app and delete functions as well.